src/test/integration/boto3-sqs/app/main.py

```diff
@@ -24,14 +24,28 @@ def run():
     )
 
     messages_copy = None
+
     response = client.receive_message(QueueUrl=queue.url, MaxNumberOfMessages=1)
+
+    # TODO Check if client.receive_message alwauys returns a list, even when there is one message
     if response:
         if "Messages" in response and response["Messages"]:
             print(f"Before first iteration: {get_current_span()}")
 
             message = response["Messages"][0]
             print(f"After access by index: {get_current_span()}")
 
+            # Restore the client.receive_message span context
+            # for context-manager calls using the messages
+            try:
+                with (message := response["Messages"]):
```

Comment:
It will work only on python3.8,
don't we want to support 3.7 (which is not EOL)?

Reply:
Yeah, the walrus is 3.8

Reply:
Done in ce7e4d51683442078625586fffcc48e86af660d6